File: src/simulation/common/startSimulation.java
package simulation.common;

import java.io.IOException;
import simulation.agent.behaviors.OnBoundaryBehaviorAttraction;
import simulation.agent.behaviors.SwarmBehavior;
import simulation.agent.behaviors.SwarmBehaviorNode;
import simulation.field.grassField.Field;

public class startSimulation {

public static void main(String[] args) throws IOException{
	
	Field f = new Field(100, // amount of agents
						500, 500, //length, width 
						1,//larger, things drawn bigger  
						new SwarmBehavior(false, .999f), //pick a behavior for the swarm 
						new OnBoundaryBehaviorAttraction(), //pick an edge behavior
						false //draw GUI?
						); 
	
	new Thread(f).start();
}
}

File: src/simulation/startSimulation.ts
import { Field } from './field/grassField/Field';
import { SwarmBehavior } from './agent/behaviors/SwarmBehavior';
import { OnBoundaryBehaviorAttraction } from './agent/behaviors/OnBoundaryBehaviorAttraction';
import { SwarmBehaviorNode } from './agent/behaviors/SwarmBehaviorNode';

async function main() {
  const field = new Field(
    100,
    500, 500,
    new SwarmBehaviorNode(true), 
    new OnBoundaryBehaviorAttraction()
  );

  console.log('Starting swarm simulation...');
  const MAX_STEPS = 100;
  for (let i = 0; i < MAX_STEPS; i++) {
    field.step();
    if (i % 10 === 0) {
      console.log(`Step ${i}: ${field.getPercentGrassCut().toFixed(2)}% grass cut`);
    }
  }
  console.log('Simulation complete.');
}

main().catch(console.error);
